src/mlpack/tests/q_learning_test.cpp

```diff
@@ -504,4 +506,45 @@ BOOST_AUTO_TEST_CASE(CartPoleWithCategoricalDQN)
   BOOST_REQUIRE(converged);
 }
 
+//! Test SAC on Pendulum task.
+BOOST_AUTO_TEST_CASE(PendulumWithSAC)
+{
+  // It isn't guaranteed that the network will converge in the specified number
+  // of iterations using random weights.
+  bool converged = false;
+  for (size_t trial = 0; trial < 3; ++trial)
+  {
+    Log::Debug << "Trial number: " << trial << std::endl;
+    // Set up the policy and replay method.
+    RandomReplay<Pendulum> replayMethod(32, 10000);
+
+    TrainingConfig config;
+    config.StepSize() = 0.001;
+    config.TargetNetworkSyncInterval() = 1;
+    config.UpdateInterval() = 3;
+
+    FFN<EmptyLoss<>, GaussianInitialization>
+        policyNetwork(EmptyLoss<>(), GaussianInitialization(0, 0.1));
+    policyNetwork.Add(new Linear<>(3, 128));
+    policyNetwork.Add(new ReLULayer<>());
+    policyNetwork.Add(new Linear<>(128, 1));
+    policyNetwork.Add(new TanHLayer<>());
+
+    FFN<EmptyLoss<>, GaussianInitialization>
+        qNetwork(EmptyLoss<>(), GaussianInitialization(0, 0.1));
+    qNetwork.Add(new Linear<>(3+1, 128));
+    qNetwork.Add(new ReLULayer<>());
+    qNetwork.Add(new Linear<>(128, 1));
+
+    // Set up Soft actor-critic agent.
+    SAC<Pendulum, decltype(qNetwork), decltype(policyNetwork), AdamUpdate>
+        agent(config, qNetwork, policyNetwork, replayMethod);
+
+    converged = testAgent<decltype(agent)>(agent, -1000, 500, 10);
```

Comment:
Also, I was thinking we might be able to make this tests a bit more robust by playing with the numbers a little. I'll run some tests tomorrow to see how much we could push it. For instance, it seems to pass the tests with `-900` while being about 10 seconds slower each run

Reply:
-900 seems like a good place, any more and it starts taking too long

Reply:
Let's try that.